### ISSUE TITLE
Use go-yaml's UnmarshalStrict to emit errors for user typos in cluster.yaml

### ISSUE THREAD

**Issue by mumoshu** [kubernetes-incubator/kube-aws#1143]
UnmarshalStrict has been introduced to go-yaml last year: https://github.com/go-yaml/yaml/pull/262

kube-aws has been implementing our own logic to emit errors in such way with [`UnknownKeys`
](https://github.com/kubernetes-incubator/kube-aws/search?utf8=%E2%9C%93&q=unknownkeys&type=) as [suggested originally in go-yaml](https://github.com/go-yaml/yaml/issues/136#issuecomment-213658588).

But honestly, it isn't/never have been sustainable 😃
Basically we should have UnknownKeys in every struct we use, which isn't achieved yet.

Just use `UnmarshalStrict` so that we never need to deal with this problem.

**Comment by Matei207** [comment]
Hi! I've been working on this issue for the past few days but I keep running into problems. The latest is that with the way the `cluster.yaml` file is parsed.

The `config` file is parsed more than once, in different contexts, from what I've seen. Before `UnmarshalStrict`, unrecognised fields would just be ignored, however, that's not possible anymore. In the case of `controlplane`, for example, `worker` cannot be parsed, the following error is thrown:

```
Error: file cluster.yaml: failed to parse cluster: yaml: unmarshal errors:
  line 132: field worker not found in type config.Cluster
```
When `etcd`, `controlplane`, etc parse the `cluster.yaml` to get their data, anything thats unrecognised in discarded. With `UnmarshalStrict` the program now panics and there is no way to simply ignore the unrecognised keys / fields. To make it work, I think a refactor would be needed (and I will be opening a separate issue for that)

@mumoshu
